packages/medusa/src/api/routes/admin/products/create-variant.ts

```diff
@@ -111,18 +169,139 @@ export default async (req, res) => {
     req.body
   )
 
+  const inventoryService: IInventoryService =
+    req.scope.resolve("inventoryService")
+  const productVariantInventoryService: ProductVariantInventoryService =
+    req.scope.resolve("productVariantInventoryService")
   const productVariantService: ProductVariantService = req.scope.resolve(
     "productVariantService"
   )
-  const productService: ProductService = req.scope.resolve("productService")
+
+  const createdId: Record<string, string | null> = {
+    variant: null,
+    inventoryItem: null,
+  }
 
   const manager: EntityManager = req.scope.resolve("manager")
   await manager.transaction(async (transactionManager) => {
-    return await productVariantService
-      .withTransaction(transactionManager)
-      .create(id, validated as CreateProductVariantInput)
+    const inventoryServiceTx =
+      inventoryService?.withTransaction?.(transactionManager)
+    const productVariantInventoryServiceTx =
+      productVariantInventoryService?.withTransaction?.(transactionManager)
```

Comment:
thought: don't think the `?`s are necessary here - we will always register a `productVariantInventoryService`